components/dashboard/src/projects/ConfigureProject.tsx

```diff
@@ -93,52 +125,121 @@ export default function () {
+        <div className="flex bg-gray-50 dark:bg-gray-800 border-b border-gray-200 dark:border-gray-600 px-6 pt-3">
           <TabMenuItem name=".gitpod.yml" selected={selectedEditor === '.gitpod.yml'} onClick={() => setSelectedEditor('.gitpod.yml')} />
-          <TabMenuItem name=".gitpod.Dockerfile" selected={selectedEditor === '.gitpod.Dockerfile'} onClick={() => setSelectedEditor('.gitpod.Dockerfile')} />
-        </div>}
+          {!!dockerfile && <TabMenuItem name=".gitpod.Dockerfile" selected={selectedEditor === '.gitpod.Dockerfile'} onClick={() => setSelectedEditor('.gitpod.Dockerfile')} />}
+        </div>
+        {editorMessage}
         <Suspense fallback={<div />}>
           {selectedEditor === '.gitpod.yml' &&
-            <MonacoEditor classes="mt-4 w-full h-64" disabled={isEditorDisabled} language="yaml" value={gitpodYml} onChange={setGitpodYml} />}
+            <MonacoEditor classes="w-full flex-grow" disabled={isEditorDisabled} language="yaml" value={gitpodYml} onChange={setGitpodYml} />}
           {selectedEditor === '.gitpod.Dockerfile' &&
-            <MonacoEditor classes="mt-4 w-full h-64" disabled={isEditorDisabled} language="dockerfile" value={dockerfile} onChange={setDockerfile} />}
+            <MonacoEditor classes="w-full flex-grow" disabled={isEditorDisabled} language="dockerfile" value={dockerfile} onChange={setDockerfile} />}
         </Suspense>
-        <div className="mt-2 flex justify-center space-x-2">
-          <button disabled={isEditorDisabled} onClick={buildProject}>Save &amp; Test Configuration</button>
-        </div>
+        {isDetecting && <div className="absolute h-full w-full flex items-center justify-center space-x-2">
+          <img className="h-5 w-5 animate-spin" src={isDark ? SpinnerDark : Spinner} />
+          <span className="font-semibold text-gray-400">Detecting project type ...</span>
+        </div>}
       </div>
-      <div className="flex-1 m-8">
-        <h3 className="text-center">Output</h3>
-        {!!workspaceCreationResult && <PrebuildLogs workspaceId={workspaceCreationResult.createdWorkspaceId} />}
+      <div className="flex-grow h-96 rounded-xl overflow-hidden bg-gray-100 dark:bg-gray-700 flex flex-col">
+        <div className="flex-grow flex">{workspaceCreationResult
+          ? <PrebuildLogs workspaceId={workspaceCreationResult.createdWorkspaceId} onInstanceUpdate={onInstanceUpdate} />
+          : <div className="flex-grow flex flex-col items-center justify-center">
+              <img className="w-14" role="presentation" src={isDark ? PrebuildLogsEmptyDark : PrebuildLogsEmpty} />
+              <h3 className="text-center text-lg text-gray-500 dark:text-gray-50 mt-4">No Recent Prebuild</h3>
+              <p className="text-center text-base text-gray-500 dark:text-gray-400 mt-2 w-64">Edit the project configuration on the left to get started. <a className="learn-more" href="https://www.gitpod.io/docs/config-gitpod-file/">Learn more</a>.</p>
```

Comment:
I'd feel a bit strange with the proposed new copy:

> Prebuilds run on every commit

As a user discovering the Project Configurator, I'm now faced with a conflict: Should I click the "Run Prebuild" button, or push new commits instead as suggested here?

I don't think commits are relevant here, and even add confusion (either you're in a workspace, pushing commits, or you're in the Project Configurator and you test a config with a few clicks).

Reply:
You are probably right here! The intent here was to _refrain_ from a) having duplicate content as the auto-detection message on the left[[1](https://github.com/gitpod-io/gitpod/pull/5236/files#diff-f4b8adb02a103d42c0de94bec0d32a6179440371ee01c21336a2181fcfa3a66dR110)][[2](https://github.com/gitpod-io/gitpod/pull/5236/files#diff-f4b8adb02a103d42c0de94bec0d32a6179440371ee01c21336a2181fcfa3a66dR118)] and b) showing an inaccurate message as editing the configuration is not possible for projects with a git-based configuration already in place. Instead this could help educate users about prebuilds.

What do you think of:

1. Keeping the following on the empty state on the right:

> Run prebuild or open a new workspace to edit project configuration.

2. Changing the message on the left when the project has been detected successfully to something like the following:

> The starter template below has been added based on the project type detected.

🍊 🍊 🍊 🍊 

Also, enabling the _Run Prebuild_ button would be great here, right?